kernel/names.ml

```diff
@@ -841,22 +842,36 @@ struct
 
     let arg c = c.proj_arg
 
-    let equal a b =
-      eq_ind a.proj_ind b.proj_ind && Int.equal a.proj_arg b.proj_arg
+    let relevant c = c.proj_relevant
 
     let hash p =
       Hashset.Combine.combinesmall p.proj_arg (ind_hash p.proj_ind)
 
+    let bcomp b1 b2 = match b1, b2 with
+    | true, true | false, false -> 0
+    | true, false -> 1
+    | false, true -> -1
+
+    let compare_gen a b =
+      let c = Int.compare a.proj_arg b.proj_arg in
+      if c <> 0 then c
+      else
+        let c = Int.compare a.proj_npars b.proj_npars in
+        if c <> 0 then c
+        else
+          let c = Label.compare a.proj_name b.proj_name in
+          if c <> 0 then c
+          else bcomp a.proj_relevant b.proj_relevant
```

Comment:
Can't you ignore proj_npars proj_name and proj_relevant here?

Reply:
Why would you? This could be a source of unsoundness if not used correctly. I think that this is a dangerous game.